corehq/apps/app_manager/management/commands/migrate_to_conditional_case_update.py

```diff
@@ -0,0 +1,85 @@
+                            usercase_update_action['update'] = wrap_transition_from_old_update_case_action(
+                                usercase_update_action['update']
+                            )
+                        if subcases_action:
+                            # Transition form.actions.subcases.name_path
+                            if subcases_action.get('case_name', ''):
+                                case_name = subcases_action['case_name']
+                                subcases_action['case_name'] = get_new_case_update_json(case_name)
+                                del subcases_action['case_name']
+                            # Transition form.actions.subcases.case_properties
+                            if subcases_action.get('case_properties', ''):
+                                subcases_action['case_properties'] = wrap_transition_from_old_update_case_action(
+                                    subcases_action['case_properties'])
+                # Advanced module
+                elif module['module_type'] == "advanced":
+                    for form in module['forms']:
+                        if form['form_type'] == 'advanced_form':
+                            actions = form.get('actions', '')
+                        elif form['form_type'] == 'shadow_form':
+                            actions = form.get('extra_actions', '')
+                        if actions:
+                            open_cases_action = actions.get('open_cases', '')
+                            # Get first elem in list if exists
+                            load_update_action = next(iter(actions.get('load_update_cases', '')), None)
+                            if open_cases_action:
+                                # Transition form.actions.open_cases_action.name_path
+                                if open_case_action.get('name_path', ''):
+                                    name_path = open_case_action['name_path']
+                                    open_case_action['name_update'] = get_new_case_update_json(name_path)
+                                    del open_case_action['name_path']
+                                # Transition form.actions.open_cases_action.case_properties
+                                if open_case_action.get('case_properties', ''):
+                                    open_case_action['case_properties'] = wrap_transition_from_old_update_case_action(
+                                        open_case_action['case_properties'])
+                            if load_update_action:
+                                # Transition form.actions.load_update_cases.case_properties
+                                if load_update_action.get('case_properties', ''):
+                                    load_update_action['case_properties'] = wrap_transition_from_old_update_case_action(
+                                        load_update_action['case_properties'])
+        return Application.wrap(app_doc)
```

Comment:
Is every document going to be updated, even those that don't use case management? If not, I'd keep track of whether or not the doc was "dirtied" and return None if not, so you don't do unnecessary saves.

Reply:
Also, this should use `wrap_app` so that linked apps and deleted apps get wrapped correctly.